src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -348,7 +370,21 @@ def _initialize_aggr_function(self, flow_obj: object):
                 )
             if not hasattr(aggr_func, "__original_function"):
                 aggr_func = _traced(aggr_func)
-            self._aggr_func = aggr_func
+            if inspect.iscoroutinefunction(aggr_func):
+
+                def run_async_function_sync(*args, **kwargs):
+                    return async_run_allowing_running_loop(aggr_func, *args, **kwargs)
+
+                self._aggr_func = run_async_function_sync
+                self._aggr_func_async = aggr_func
+            else:
+
+                async def run_sync_function_async(*args, **kwargs):
+                    with ThreadPoolExecutorWithContext() as executor:
+                        return executor.submit(aggr_func, *args, **kwargs).result()
```

Comment:
please use `await run_in_loop` to avoid blocking the world.

Reply:
Fixed.